docs/spec/slashing/state-machine.md

```diff
@@ -0,0 +1,174 @@
+# State Machine Interaction Overview
+
+## Conceptual overview
+
+### States
+
+At any given time, there are any number of validator candidates registered in the state machine.
+Each block, the top `n` candidates who are not jailed become *bonded*, meaning that they may propose and vote on blocks.
```

Comment:
clarify what `n` means - if that is the maximum bonded validators param then we should say this